front/app/modules/commercial/representativeness/admin/hooks/useReferenceData.ts

```diff
@@ -29,11 +32,15 @@ function useReferenceData(
   const [referenceDataUploaded, setReferenceDataUploaded] = useState<
     boolean | undefined
   >();
-
+  const { data: usersByAge } = useUsersByAge({ project: projectId });
+  const { data: usersByGender } = useUsersByGender({ project: projectId });
+  const { data: usersByCustomField } = useUsersByCustomField({
+    project: projectId,
+    id: userCustomField.id,
+  });
```

Comment:
Ah, but here it's a real problem. It's completely possible that you only have reference data for `usersByAge`, but not for `usersByGender` or other custom fields. Yet here you will still fetch all that data unnecessarily. I think it could be easily fixed by adding an `enabled` attribute to the props of these hooks, and passing that to the underlying `useQuery` call. You can then set `enabled` only to true if the `code` of the `IUserCustomFieldData` matches the hook. Let me know if that makes sense

Reply:
Also, we should not be calling `useUsersByCustomField` if we're already calling the specific hook (e.g. `useUsersByGender`. Otherwise we're either requesting the same data twice (and I'm not even sure if the backend will handle this well)

Reply:
I introduced a mandatory `enabled` prop for each of the hooks so that this issue can be resolved. I opted to make it mandatory, instead of optional and true by default, just to make it very explicit that this mechanism is in place.